### ISSUE TITLE
Resolve properties from parent container

### ISSUE THREAD

**Issue by vavvolo** [Swinject/Swinject#131]
Given this code

```
let parentContainer = Container()
let loader = JsonPropertyLoader(bundle: .mainBundle(), name: "properties")
try! parentContainer.applyPropertyLoader(loader)

parentContainer.register(MyServiceTypeA.self) { r in 
  MyServiceA(param: r.property("param"))
}

let childContainer = Container(parent: parentContainer)

childContainer.register(MyServiceTypeB.self) { r in 
  MyServiceM(param: r.property("param"))
}
```

The resolution of property `param` fails on the child container, I need to call `applyPropertyLoader` on both parent and child containers to make it work.

I think it would be useful to try to resolve the property using the parent container.

On master the code related to property loading has been moved to a new repo and is now an extension on `ResolverType` with no access to the parent container.

What do you think? Not sure it's worth to make a PR on the 1.x branch.

Regards

Vale

**Comment by yoichitgy** [comment]
Thanks @vavvolo for your idea😃

I moved property loader feature to [SwinjectPropertyLoader](https://github.com/Swinject/SwinjectPropertyLoader) repo because I think the loader is a kind of parser and parsing values is not responsibility of `Container`. Instead, the loader (parser) can be captured by factory closure like following:

```
let container = Container()
let loader = SomeLoader(bundle: .mainBundle(), name: "properties")

container.register(MyServiceTypeA.self) { _ in 
    MyServiceA(param: loader.property("param"))
}
```

I want to keep `Container` simpler for easier maintenance and enhancement because development resource is limited.

I can merge your PR to v1 branch, but it is appreciated if you send a PR to [SwinjectPropertyLoader](https://github.com/Swinject/SwinjectPropertyLoader) repo. In the repo, the extension to `ResolverType` is just temporal and my lazy implementation for compatibility with the property loader feature in v1 series. It is great if you implement the property loader feature on `Container` with the parent resolution👍👍👍

**Comment by vavvolo** [comment]
I'm happy to help but I'm not sure I understood your example correctly.

It's not clear to me how the parent resolution on `Container` would work given that the property loader has the properties.

Also at the moment we can register multiple property loaders and use the same resolver to get the values whilst if we capture the loader we need to refer to the right loader in the closure.

**Comment by yoichitgy** [comment]
My example code above is just telling any parse or loader can be used with a simple `Container` without our PropertyLoader classes, (though they are of course very convenient!)

The current implementation of SwinjectPropertyLoader is on ResolverType (of Swinject v2 beta), but it might be implemented as an extension to Container because #139 might simplify the problem I couldn't implement it on Container. So if you implement the parent resolution in v1 branch, it looks possible to apply the same thing to Swinject v2 with SwinjectPropertyLoader repo.

I also do not have good idea to implement the feature. We need some investigation.

**Comment by yoichitgy** [comment]
Let's discuss here as an issue for SwinjectPropertyLoader project:
https://github.com/Swinject/SwinjectPropertyLoader/issues/8
